db/commitlog/commitlog.cc

```diff
@@ -1010,6 +1011,8 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
 
         replay_position rp(_desc.id, position_type(off));
 
+        co_await with_scheduling_group(_segment_manager->cfg.sched_group, [&]() -> future<> {
+
```

Comment:
This will cause an extra allocation. There's coroutine::switch_to which doesn't (still causes a task switch, but that's life).

Please measure perf_simple_query --mode write before and after.